### PR TITLE
Recent call views

### PR DESCRIPTION
**Pull request by pulkonet**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -54,7 +55,9 @@
             android:theme="@style/AppDialogTheme">
             <intent-filter>
                 <action android:name="android.intent.action.CALL_PRIVILEGED" />
+
```
Comment (samagra14) [inline app/src/main/AndroidManifest.xml]: Why these blank spaces within indent filters?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -66,7 +69,8 @@
             </intent-filter>
         </activity>
         <activity android:name=".Settings" />
-        <activity android:name=".LoginActivity"></activity>
+        <activity android:name=".LoginActivity" />
+        <activity android:name=".RecentCalls"></activity>
     </application>
 
 </manifest>
```
Comment (samagra14) [inline app/src/main/AndroidManifest.xml]: Leave a blank line at EOF.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java [hunk 3]
```diff
@@ -23,7 +23,7 @@
     private static final String CALL_TIME = "time";
     private static final String CALL_TYPE= "type";
     private static final String MESSAGE="message";
-
+    private static final String CALL_DURATION="duration";
     private static final String TABLE_login= "login";
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: change TABLE_login to TABLE_LOGIN.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java [hunk 4]
```diff
@@ -135,4 +136,24 @@ public String getUser(String name) {
             cursor.moveToFirst();
         return cursor.getString(1);
     }
+
+    public List<CallerDetails> getAllCallsByNumber(String[] number){
+        List<CallerDetails> userList = new ArrayList<>();
+        String query = "SELECT * FROM " + TABLE_CALLERS+ " WHERE "+ CALLER_NUMBER+ "=?";
+        SQLiteDatabase db = this.getReadableDatabase();
+        Cursor cursor = db.rawQuery(query, number); if (cursor.moveToFirst()) {
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: Using rawQuery method should be avoided. Use query method with proper selection Args.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java [hunk 5]
```diff
@@ -135,4 +136,24 @@ public String getUser(String name) {
             cursor.moveToFirst();
         return cursor.getString(1);
     }
+
+    public List<CallerDetails> getAllCallsByNumber(String[] number){
+        List<CallerDetails> userList = new ArrayList<>();
+        String query = "SELECT * FROM " + TABLE_CALLERS+ " WHERE "+ CALLER_NUMBER+ "=?";
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: Use selection Arguements.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java [hunk 6]
```diff
@@ -135,4 +136,24 @@ public String getUser(String name) {
...
+                userList.add(cd);
+            } while (cursor.moveToNext());
+        }
+        Collections.reverse(userList);
+        return userList;
+    }
+    public String getName(String[] number){
+        String query = "SELECT * FROM " + TABLE_CALLERS+ " WHERE "+ CALLER_NUMBER+ "=?";
+        SQLiteDatabase db = this.getReadableDatabase();
+        Cursor cursor = db.rawQuery(query, number);
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: Too many rawQueries can have performance issues. Use of query() is always a better choice.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        Toast.makeText(getApplicationContext(),number[0],Toast.LENGTH_LONG).show();
+        recyclerView = (RecyclerView) findViewById(R.id.recycler_caller_details);
+        dataBaseHandler = DataBaseHandler.getInstance(getApplicationContext());
+        addToList();
+       // nameView.setText(dataBaseHandler.getName(number));
+        //nameView is currently not working
+    }
+
+    /*This method generates a list that has calls from a particular number*/
+    private List<CallerDetails> createList() {
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Rename this function to createCallListFromNumber. createList is a vague name.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        List<CallerDetails> result;
+        result=dataBaseHandler.getAllCallsByNumber(number);
+        Log.e("result",result+"");
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Please not in error stream. Please.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.List;
+
+import jp.wasabeef.recyclerview.animators.adapters.ScaleInAnimationAdapter;
+
+public class RecentCalls extends AppCompatActivity {
+
+    private DataBaseHandler dataBaseHandler;
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Add a static and final log tag.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        List<CallerDetails> result;
+        result=dataBaseHandler.getAllCallsByNumber(number);
+        Log.e("result",result+"");
+        return result;
+    }
+    /*THis method sets the adapter for recycler view and also adds animation to the card*/
+    public void addToList(){
+
+        recyclerView.setHasFixedSize(true);
+        LinearLayoutManager llm = new LinearLayoutManager(getApplicationContext());
```
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Even if llm is a local variable.Still use full names.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java (hunk 5 above)
Comment (pulkonet) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: What? I found this to be quite convincing.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java (hunk 10 above)
Comment (pulkonet) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Checked.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java (hunk 8 above)
Comment (pulkonet) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Checked.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java (hunk 7 above)
Comment (pulkonet) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Checked.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java (hunk 9 above)
Comment (pulkonet) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: Why?

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java (hunk 5 above)
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: For now it is okay but it is not a good practice yr. Rawquery method is too slow.
See this
https://stackoverflow.com/questions/8830753/android-sqlite-which-query-query-or-rawquery-is-faster

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java (hunk 3 above)
Comment (pulkonet) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/DataBaseHandler.java]: Checked.

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (pulkonet) [inline app/src/main/AndroidManifest.xml]: Checked.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (pulkonet) [inline app/src/main/AndroidManifest.xml]: They were already there. Checked tho.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java (hunk 9 above)
Comment (samagra14) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/RecentCalls.java]: @pulkonet As I said log messages should be directed to debug or info stream. But there can be a large number of debug messages. So it becomes easy to search a particular log message with the help of activity specific log tags.

**Pull request merged**